prometheus/collectors/go_collector_latest_test.go

```diff
@@ -62,6 +62,28 @@ var memstatMetrics = []string{
 	"go_memstats_sys_bytes",
 }
 
+func withDefaultRuntimeMetrics(metricNames []string, withoutGC, withoutSched bool) []string {
```

Comment:
I see your point. I also would like for it to be simpler, and maybe there are ways to simplify it that I'm not seeing right now... but in any case it's not only about deduplicating and sorting; it's also about not including metrics (especially when they don't exist in go1.20) according to the different allow/deny list scenarios. In fact it's also used once in `TestGoCollectorDenyList`. Shall I change the name to `dedupSort` all the same? Or make any other change?

Reply:
And thanks for taking care of the other 2 problems! <3

Reply:
> . but in any case it's not only about deduplicating and sorting; it's also about not including metrics (especially when they don't exist in go1.20)

but that is solved with `defaultRuntimeMetrics` generation, no?

Reply:
The `defaultRuntimeMetrics` generation solves the problem of defining what the set of default runtime metrics looks like for a given Go version; in practice this allows adding the right set of default runtime metrics as is, when there are no deny lists at all OR when no deny list matches either all `gc` or all `sched` metrics. 

But in this function (which lives in a file that has to work with all the supported Go versions indifferently) I chose the approach where in practice I append the non-denied default metrics to the rest of the other requested metrics in a hard-coded way. That is to say, I append a subset of those metrics; a subset that is not dynamically generated, unlike the whole set.

So when it comes to cases when only some of the default runtime metrics have to be filtered out because of deny lists matching `gc` OR `sched` (not both: both is a non-problem), the subset to be appended may be different according to the version but I cannot dynamically/automatically reflect those differences in that function: hence all the many conditionals. Shall I think of a different, more dynamic solution?

Reply:
Incidentally: Arthur just wrote in Slack that we shouldn’t be worrying about supporting go1.20. If that means that the tests for go1.20 can be deleted, then at least half of this problem is solved, as the default runtime metrics list looks exactly the same starting from go 1.21. But maybe there are reasons we should still keeping them around for a while.

Reply:
I'm fine with deleting 1.20 test files, we don't even run them in CI anymore